### ISSUE TITLE
Very interesting, is there a publication available?

### ISSUE THREAD

**Issue by tasansal** [rhoadesScholar/RoSE#1]
Hello!

I just ran into your repo while looking at https://github.com/lucidrains/rotary-embedding-torch in PyPI :) 

This method you have here is very interesting, and I would like to learn more and see comparisons. Is there a publication you can share? (I searched around but couldn't find one).

Cheers

**Comment by rhoadesScholar** [comment]
Hi @tasansal!

Thank you for your interest :) Unfortunately there is not yet a publication.

What applications do you have in mind? I'd be happy to put together some comparisons for you.

Cheers,
Jeff

**Comment by tasansal** [comment]
Hi, thanks for your reply!

So we are doing some pre-training for a certain type of 3D scientific data (exploration seismology). We have a few challenges:

* Every different survey has a different sensor configuration. e.g. the spacing of the pixels may vary, similar to MRI equipment. We are, again, not handling this and wasting model capacity to learn this. I believe this is where RoSE can potentially help.
* Our images are very large (e.g. 10,000 x 15,000 x 4,000) pixels. Every pixel has a physical meaning (geographic coordinate x/y and depth: z). When we train, we have to subset this with 3D sliding windows to be able to fit into GPU memory. When we do that (e.g. get a 256x256x256 subset) we lost the large context position. I'm not sure if RoSE would be beneficial here but we could define the full domain with RoPE or RoSE and handle subsetting in that as well.

**Comment by rhoadesScholar** [comment]
That’s remarkably similar to work I’m currently doing with FIBSEM data — also teravoxel-scale volumetric datasets where maintaining large-scale context is critical. I’ve been working on tackling exactly the loss of “large context position” issue you mentioned, and RoSE is one component of a transformer I’ve been developing to address it.

Would you be interested in collaborating? I think a solution that works well for both geographical and biological volumetric data could generalize much better than something developed in isolation for just one domain.

**Comment by L-Reichardt** [comment]
Very interesting work. Is this a nD generalized version based on [mixed-2D-RoPE](https://arxiv.org/html/2403.13298v1)?
